src/mpid/ch4/netmod/ucx/ucx_am_impl.h

```diff
@@ -238,4 +266,202 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_UCX_do_am_isend_bulk(int rank, MPIR_Comm * co
     goto fn_exit;
 }
 
+MPL_STATIC_INLINE_PREFIX int MPIDI_UCX_do_am_isend_pipeline(int rank, MPIR_Comm * comm,
```

Comment:
Hm, this function is ~200 lines long. Instead of an `issue_deferred` argument, can the logic be split into 2 functions?

Reply:
Yeah, the function does the double duty 1) attempt issuing the first segment; 2) issuing more segments for deferred operation. These two functions share about 90% of their logic, so I combined them together. I thought this would be easier.

Let me see if I can split the functionalities (packing, sending, immediate complete handling) out so the main logic is cleaner.